unit4/generators.py

```diff
@@ -0,0 +1,91 @@
+from typing import Generator
+import itertools
+################################## 4.1.2 ##################################
+
+
+def translate(sentence: str) -> str:
+    """
+    translates a sentence from spanish to english
+    :param sentence: original sentence
+    :return: the translated sentence
+    """
+    words: dict[str] = {'esta': 'is', 'la': 'the', 'en': 'in', 'gato': 'cat', 'casa': 'house', 'el': 'the'}
```

Comment:
```suggestion
    words: dict[str, str] = {'esta': 'is', 'la': 'the', 'en': 'in', 'gato': 'cat', 'casa': 'house', 'el': 'the'}
```
When you use type hints with a dictionary, refer to both key and value types